base/common/src/main/java/org/dogtagpki/nss/NSSDatabase.java

```diff
@@ -333,17 +356,25 @@ public void addPEMCertificate(
             cmd.add("-d");
             cmd.add(path.toString());
 
-            // TODO: Add support for HSM.
-            String password = passwordStore.getPassword("internal", 0);
+            if (tokenName != null) {
+                cmd.add("-h");
+                cmd.add(tokenName);
+            }
 
-            if (password != null) {
-                passwordPath = Files.createTempFile("nss-password-", ".txt", FILE_PERMISSIONS);
-                logger.info("Storing password into " + passwordPath);
+            if (passwordStore != null) {
+
+                tokenName = tokenName == null ? "internal" : "hardware-" + tokenName;
```

Comment:
Do we want to destructively modify `tokenName` here? I'd perhaps stick it in a new variable so we leave `tokenName` (as passed in) alone, in case we need it again later for something.

Reply:
OK, fixed.